lib/checks/isQuadratic.js

```diff
@@ -0,0 +1,45 @@
+'use strict';
+
+const Node = require('../node');
+
+// TODO(ael)
+function isQuadratic(node) {
+  if (!Node.Type.isOperator(node) || node.op !== '+') {
+    return false;
+  }
+
+  if (node.args.length > 3) {
+    return false;
+  }
+
+  const secondDegreeTerms = node.args.filter(isSecondDegree);
+  const firstDegreeTerms = node.args.filter(isFirstDegree);
+  const constantTerms = node.args.filter(Node.Type.isConstant);
+
+  if (secondDegreeTerms.length !== 1 || firstDegreeTerms.length > 1 ||
+    constantTerms.length !== 1) {
```

Comment:
hmm why'd you choose this combination?

I agree that there should be a secondDegreeTerm for sure

but what about `x^2 + x`? that seems like a quadratic, but there's no constant term